apps/web/middleware.ts

```diff
@@ -70,6 +88,25 @@ const middleware: NextMiddleware = async (req) => {
     requestHeaders.set("x-csp-enforce", "true");
   }
 
+  if (isValidOrgDomain) {
+    // Match /:slug to determine if it corresponds to org subteam slug or org user slug
+    const [first, slug, ...rest] = url.pathname.split("/");
+    // In the presence of an organization, if not team profile, a user or team is being accessed
+    if (first === "" && rest.length === 0) {
+      // Fetch the corresponding subteams for the entered organization
+      const getSubteams = await fetch(`${WEBAPP_URL}/api/organizations/${currentOrgDomain}/subteams`);
```

Comment:
This fetch can add ~2 seconds to the profile/team page response time on cold starts, can we get this teams using a function instead of fetch the endpoint?

Reply:
This is not possible, we can use prisma on middleware because we are using the prisma data proxy, but we can't force to selfhoster to used so here can left this techdebt until we get support ISR on api endpoints

Reply:
Thanks for the update. Opening PR to be merged then.